pkg/openwsn/contrib/radio_hal.c

```diff
@@ -0,0 +1,316 @@
+/*
+ * Copyright (C) 2020 Inria
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     pkg_openwsn
+ * @{
+ *
+ * @file
+ * @brief       RIOT adaption of the "radio" bsp module
+ *
+ * @author      Francisco Molina <francois-xavier.molina@inria.fr>
+ * @}
+ */
+#include <sys/uio.h>
+
+#include "leds.h"
+#include "debugpins.h"
+#include "sctimer.h"
+#include "idmanager.h"
+#include "eui64.h"
+
+#include "byteorder.h"
+
+#include "luid.h"
+#include "net/ieee802154.h"
+#include "net/ieee802154/radio.h"
+
+#include "openwsn.h"
+#include "openwsn_radio.h"
+
+#define LOG_LEVEL LOG_NONE
+#include "log.h"
+
+openwsn_radio_t openwsn_radio;
+
+/* stores the event capture time */
+static PORT_TIMER_WIDTH _txrx_event_capture_time = 0;
+
+void _idmanager_addr_override(void)
+{
+    /* Initiate Id manager here and not in `openstack_init` function to
+       allow overriding the short id address before additional stack
+       components are initiated */
+    idmanager_init();
+
+    /* override 16b address to avoid short address collision */
+    network_uint16_t short_addr;
+    luid_get_short(&short_addr);
+    open_addr_t id;
+    id.type = ADDR_16B;
+    memcpy(&id.addr_16b, short_addr.u8, IEEE802154_SHORT_ADDRESS_LEN);
+    idmanager_setMyID(&id);
+
+    /* override PANID */
+    id.type = ADDR_PANID;
+    network_uint16_t panid_be = byteorder_htons(OPENWSN_PANID);
+    memcpy(&id.addr_16b, &panid_be, IEEE802154_SHORT_ADDRESS_LEN);
+    idmanager_setMyID(&id);
+
+    /* recover ADDR_64B */
+    eui64_t eui64;
+    eui64_get(eui64.uint8);
+
+    /* Set all IEEE addresses */
+    uint16_t panid = OPENWSN_PANID;
+    ieee802154_radio_set_hw_addr_filter(openwsn_radio.dev, &short_addr,
+                                        &eui64, &panid);
+
+}
+
+static void _hal_radio_cb(ieee802154_dev_t *dev, ieee802154_trx_ev_t status)
+{
+    (void)dev;
+
+    debugpins_isr_set();
+    _txrx_event_capture_time = sctimer_readCounter();
+    debugpins_isr_clr();
+
+    switch (status) {
+    case IEEE802154_RADIO_CONFIRM_TX_DONE:
+        ieee802154_radio_request_set_trx_state(openwsn_radio.dev,
```

Comment:
someone should call `ieee802154_radio_confirm_transmit()` on `TX_DONE`. Although most radios do "nothing" there, this is required by the Abstract State Machine of the Radio HAL. Some radios might use it to synchronize internal state variables

Reply:
done